analyzers/tests/SonarAnalyzer.UnitTest/TestCases/RedundancyInConstructorDestructorDeclaration.CSharp10.cs

```diff
@@ -19,3 +19,46 @@ struct Struct
 {
     public Struct() { } // Noncompliant
 }
+
+// https://github.com/SonarSource/sonar-dotnet/issues/8087
+public readonly struct Repro_8087
+{
+    // More info here https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/constructor-errors?f1url=%3FappId%3Droslyn%26k%3Dk(CS8983)#constructors-in-struct-types
+    public Repro_8087() { } // Compliant - Would mix with CS8983
+
+    public bool Foo { get; init; } = true;
+}
+
```

Comment:
We need one counter-example:
```cs
public struct Noncompliant
{
    public Noncompliant() { } // Noncompliant

    public bool Foo { get; set; } // No initializer
}
```

Reply:
I am moving this example to the CSharp11 file because I am getting this error:
```
CSharp10: Unexpected build error [CS0843]: Auto-implemented property 'Noncompliant.Foo' must be fully assigned before control is returned to the caller. Consider updating to language version '11.0' to auto-default the property. on line 34
```